### PR TITLE
[FIX] account_edi_ubl_cii: update Belgian EAS and validate the endpoint

### PR DESCRIPTION
**Pull request by JulienVR**
Replace every mentions of 9925 to 0208 for Belgian companies (public administration also have this 10 digits number). The EAS 0208 corresponds to the company_registry.

Introduce a mechanism to validate the peppol_endpoint.

See: https://openpeppol.atlassian.net/wiki/spaces/Belgium/overview

task-3297311

### CODE REVIEW

File: addons/account_edi_ubl_cii/models/res_partner.py [hunk 1]
```diff
@@ -139,36 +147,58 @@ def _compute_ubl_cii_format(self):
...
+                    try:
+                        self._validate_peppol_endpoint(self.peppol_eas, self[field])
+                        return eas  # ...until we obtain a valid one
+                    except ValidationError:
+                        continue
+
+    def _validate_peppol_endpoint(self, eas, endpoint):
+        """ Contains all the rules regarding the peppol_endpoint."""
+        if eas == '0208' and not re.match(r"^\d{10}$", endpoint):
+            raise ValidationError(_("The Peppol endpoint does not seem to be valid. The expected format is: 0239843188"))
```
Comment (smetl) [inline addons/account_edi_ubl_cii/models/res_partner.py:200]: Maybe it's time to add the siret as well here

File: addons/account_edi_ubl_cii/models/res_partner.py [hunk 2]
```diff
@@ -139,36 +147,58 @@ def _compute_ubl_cii_format(self):
...
+            new_endpoint = None
+            if partner.country_code in EAS_MAPPING:
+                field = EAS_MAPPING[partner.country_code].get(partner.peppol_eas)
+                if field and field in partner._fields and partner[field]:
+                    try:
+                        partner._validate_peppol_endpoint(partner.peppol_eas, partner[field])
+                        new_endpoint = partner[field]
+                    except ValidationError:
+                        pass
+            partner.peppol_endpoint = new_endpoint or partner.peppol_endpoint
```
Comment (smetl) [inline addons/account_edi_ubl_cii/models/res_partner.py:182]:
```suggestion
            partner.peppol_endpoint = partner.peppol_endpoint
            if partner.country_code in EAS_MAPPING:
                field = EAS_MAPPING[partner.country_code].get(partner.peppol_eas)
                if field and field in partner._fields and partner[field]:
                    try:
                        partner._validate_peppol_endpoint(partner.peppol_eas, partner[field])
                        partner.peppol_endpoint = partner[field]
                    except ValidationError:
                        pass
```

File: addons/account_edi_ubl_cii/models/account_edi_common.py [hunk 3]
```diff
@@ -44,7 +44,7 @@
     'AT': {'9915': 'vat'},
     'AU': {'0151': 'vat'},
     'BA': {'9924': 'vat'},
-    'BE': {'9925': 'vat'},
+    'BE': {'0208': 'company_registry'},
```
[the remark this answers is not in the document]
Comment (JulienVR) [reply addons/account_edi_ubl_cii/models/account_edi_common.py:48]: 9925 has been removed on the 1st September 2023 (https://openpeppol.atlassian.net/wiki/spaces/Belgium/pages/1111883917/Belgian+Service+Providers+Forum#9956%3E0208-migration) :)

File: addons/account_edi_ubl_cii/tests/test_partner_peppol_fields.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+from contextlib import contextmanager
+from mock import patch
```
Comment (smetl) [inline addons/account_edi_ubl_cii/tests/test_partner_peppol_fields.py:4]:
```suggestion
try:
    from unittest.mock import patch
except ImportError:
    from mock import patch
```

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**